sql/core/src/test/scala/org/apache/spark/sql/CollationSQLExpressionsSuite.scala

```diff
@@ -2319,6 +2319,30 @@ class CollationSQLExpressionsSuite
     )
   }
 
+  // scalastyle:off nonascii
+  test("approx_count_distinct returns correct with collation") {
+    Seq(
+      ("utf8_lcase", Seq("a", "a", "A"), Seq(Row(1))),
+      ("utf8_lcase", Seq("aCfew", "acFEw", "ACFEW"), Seq(Row(1))),
+      ("UNICODE_CI_AI", Seq("č", "c", "Ć", "C", "Z", "Ž"), Seq(Row(2))),
+      ("UNICODE_CI_AI", Seq("fAFfewfćČĆfečwćCsŠ", "fAffEWfćČCfeCwcćss"), Seq(Row(1))),
+      ("UNICODE_CI_AI", Seq("fAcfewfćČĆfečwćCsŠ", "fAffEWfćČCfeCwcćss"), Seq(Row(2)))
+    ).foreach{
+      case (collationName: String, input: Seq[String], expected: Seq[Row]) =>
+        checkAnswer(sql(
+          s"""
+             with t as (
+             select collate(col1, '$collationName') as c
+             from
+             values ${input.map(s => s"('$s')").mkString(", ")}
+          )
+          SELECT approx_count_distinct(c) FROM t
+             """), expected)
+    }
```

Comment:
Let's use the following pattern:

1. Create `case class` which encapsulates the specific check we're going to make
2. Construct sequence of instances of the previously created case class
3. Iterate the sequence and perform required checks; this is already in place, but I suggest we use `CREATE TABLE` + `INSERT` statements instead of `WITH` to make the code more readable
4. Change session-level default collation and perform the checks again to verify `approx_count_distinct` works well when non-`UTF8_BINARY` collation is session default

Example test which already follows this approach: https://github.com/apache/spark/blob/5d16c3134c442a5546251fd7c42b1da9fdf3969e/sql/core/src/test/scala/org/apache/spark/sql/CollationSQLExpressionsSuite.scala#L42-L70